src/TeaAsAService/Tea.Core/Domain/User.cs

```diff
@@ -72,5 +80,42 @@ public static User CreateLocalDevUser()
                 LastBrewTimeUtc = DateTime.UtcNow
             };
         }
+
+        static bool ValidatePassword(string password)
+        {
+            const int MIN_LENGTH = 8;
+
+            if (password == null) throw new ArgumentNullException();
+
+            bool meetsLengthRequirements = password.Length >= MIN_LENGTH;
+            bool hasUpperCaseLetter = false;
+            bool hasLowerCaseLetter = false;
+            bool hasDecimalDigit = false;
+            bool hasSpecialCharacter = false;
+
+            if (meetsLengthRequirements)

```

Comment:
Try not to use long nested if statements, they make the code hard to read. Instead of looping through each char and testing for each section we should just be able to use Linq to see if it contains what we are looking for, and if not then just exit the method with a false right away. 

For instance checking for an uppercase char would look like: